File: src/main/java/renderer/constants/BlendMode.java
package renderer.constants;

import org.lwjgl.opengl.GL11;
import org.lwjgl.opengl.GL14;

public enum BlendMode {
    FUNC_ADD(GL14.GL_FUNC_ADD);

    public final int mode;

    BlendMode(int mode) {
        this.mode = mode;
    }


    public enum Factor {
        ONE(GL11.GL_ONE);

        public final int glFactor;

        Factor(int glFactor) {
            this.glFactor = glFactor;
        }
    }
}

File: engine/src/main/kotlin/de/hanno/hpengine/graphics/renderer/constants/BlendMode.kt
package de.hanno.hpengine.graphics.renderer.constants

import org.lwjgl.opengl.GL11
import org.lwjgl.opengl.GL14

val BlendMode.mode: Int get() = when(this) {
    BlendMode.FUNC_ADD -> GL14.GL_FUNC_ADD
}

val BlendMode.Factor.glFactor: Int get() = when(this) {
    BlendMode.Factor.ZERO -> GL11.GL_ZERO
    BlendMode.Factor.ONE -> GL11.GL_ONE
    BlendMode.Factor.SRC_ALPHA -> GL11.GL_SRC_ALPHA
    BlendMode.Factor.ONE_MINUS_SRC_ALPHA -> GL11.GL_ONE_MINUS_SRC_ALPHA
}